### DISCUSSION TITLE
Conductor server and React.js

### DISCUSSION THREAD

**Discussion by Germs31** [Netflix/conductor#2751] [Q&A]
HI. Am I allowed to hit the conductor server from a react application? is this good practice? Im under the impression that it should be hit from the backend. Can anyone give me in any insight? im pretty confused. please and thank you

Front-end -----> Back-end ------> Start Conductor

**Comment by aravindanr** [comment]
The Conductor UI is a react application that hits Conductor directly. So it's possible. @peterlau if you have any thoughts, please share.

**Reply by Germs31** [reply]
Oof. i get CORS errors when trying to hit the conductor server endpoints from my react app.

**Reply by Germs31** [reply]
I can hit conductor server endpoints from my backend tho. sorry im a noob lol

**Reply by Germs31** [reply]
Conductor UI uses this endpoint when retrieving metadata workflows "/api/wfe/metadata/workflow". I was curious as to why it hits this because it is not an endpoint provided with in the documentation. @aravindanr

**Reply by peterlau** [reply]
That URL `/api/wfe/*` is associated with the legacy (2.x) UI. The latest conductor UI (V3.x) should be requesting from `/api`. In order to prevent CORS issues, we recommend setting up a reverse proxy of the conductor backend API to `/api` on the port & domain that is hosting the UI. This is done automatically (`localhost:8080 -> localhost:5000/api`) if you host the UI in debug mode using the `yarn run start` dev runner. The `/docker` directory of the repo also contains and example configuration for nginx.

**Reply by peterlau** [reply]
Alternatively, you can put the conductor server behind a CORS configured web server and modify `plugins/fetch.js` in the UI to fetch directly from your different-origin backend.

**Reply by Germs31** [reply]
@peterlau @aravindanr Thank you both so much! I will give these a try right away!

**Comment by v1r3n** [comment]
Hi,

Yes you can certainly call Conductor APIs from React backend.  

Some potential reasons to do that:

1. Mark as task as completed in a workflow with human inputs (like approve expense and it marks the approval task as completed)
2. Start a new workflow, e.g when a user adds an item to the shopping cart, start a workflow to track the purchase flow
